controllers/bootstrap/init.go

```diff
@@ -196,50 +196,76 @@ func (b *Bootstrap) CrossplaneOperatorProviderOperator(instance *apiv3.CommonSer
 
 	// Install Crossplane Operator & Provider Operator
 	bedrockshim := false
+	removeCrossplaneProvider := false
 	if instance.Spec.Features != nil {
 		if instance.Spec.Features.Bedrockshim != nil {
 			bedrockshim = instance.Spec.Features.Bedrockshim.Enabled
+			removeCrossplaneProvider = instance.Spec.Features.Bedrockshim.CrossplaneProviderRemoval
 		}
 	}
+
 	var isLater bool
 	var err error
-	if bedrockshim {
-		b.CSData.CrossplaneProvider = "odlm"
 
-		if b.SaasEnable {
-			b.CSData.CrossplaneProvider = "ibmcloud"
+	if b.MultiInstancesEnable {
+		resourceName := constant.CrossSubscription
+		isLater, err = b.CompareChannel(resourceName)
+		if err != nil {
+			return err
 		}
-		if b.MultiInstancesEnable {
-			resourceName := constant.CrossSubscription
-			isLater, err = b.CompareChannel(resourceName)
-			if err != nil {
-				return err
+	} else {
+		isLater = false
+	}
+
+	//isLater value of false means we install, true means we do not install
+	if !isLater {
+		if bedrockshim {
+			if removeCrossplaneProvider {
+				//delete Crossplane Provider Subscription
+				klog.Info("try to delete CrossplaneProvider")
+				if DeleteErr := b.DeleteCrossplaneProviderSubscription(b.CSData.ControlNs); DeleteErr != nil {
+					return DeleteErr
+				}
+			} else {
+				if updateErr := b.CreateorUpdateCFCrossplaneConfigMap("'false'"); updateErr != nil {
+					return updateErr
+				}
 			}
-		} else {
-			isLater = false
-		}
-		//isLater value of false means we install, true means we do not install
-		if !isLater {
+
+			b.CSData.CrossplaneProvider = "odlm"
+			if b.SaasEnable {
+				b.CSData.CrossplaneProvider = "ibmcloud"
+			}
+
+			// install crossplane operator
 			if err := b.installCrossplaneOperator(); err != nil {
 				return err
 			}
-			switch b.CSData.CrossplaneProvider {
-			case "odlm":
-				if err := b.installKubernetesProvider(); err != nil {
-					return err
-				}
-			case "ibmcloud":
-				if err := b.installIBMCloudProvider(); err != nil {
-					return err
+			// we install provider only if we don't need to delete it
+			if !removeCrossplaneProvider {
+				switch b.CSData.CrossplaneProvider {
+				case "odlm":
+					if err := b.installKubernetesProvider(); err != nil {
+						return err
+					}
+				case "ibmcloud":
+					if err := b.installIBMCloudProvider(); err != nil {
+						return err
+					}
 				}
 			}
 		} else {
-			klog.Infof("Crossplane operator already exists at a later version in control namespace. Skipping.")
+			// delete crossplane and provider operator if exist
+			if err := b.DeleteCrossplaneAndProviderSubscription(b.CSData.ControlNs); err != nil {
+				return err
+			}
+			value := "'" + strconv.FormatBool(removeCrossplaneProvider) + "'"
+			if updateErr := b.CreateorUpdateCFCrossplaneConfigMap(value); updateErr != nil {
+				return updateErr
+			}
```

Comment:
Wondering if we should move this part into `DeleteCrossplaneAndProviderSubscription(b.CSData.ControlNs)`, like what we did for `DeleteCrossplaneProviderSubscription`

I am considering that we should go over all CommonService CRs in the cluster and determine which value should be put into ConfigMap.

Reply:
Yes, we should move it to DeleteCrossplaneAndProviderSubscription